### ISSUE TITLE
Best way to set up authentication

### ISSUE THREAD

**Issue by roborourke** [mvertes/docker-alpine-mongo#11]
Thanks for this container - I'd ideally use it within a docker-compose based network and not have it publicly facing at all but I'm testing out zeit.co's now tool and it only supports a single Dockerfile or package.json.

The simplest approach in my mind is to add some environment variables for a user & password that can be overridden on run/deploy. If set then we can enable auth with those details.

Is this something you'd want to support directly in this repo or are there other better approaches you can think of?

**Comment by mvertes** [comment]
Yes, authentication could be setup using environment variables and a procedure in the existing startup script `run.sh`. This is something I would gladly support.

**Comment by roborourke** [comment]
Awesome! I can make a PR later this week unless you know how you'd do it already.

**Comment by mvertes** [comment]
Ok, please proceed. I do not have it ready.

**Comment by wupeaking** [comment]
Error relocating /usr/bin/mongod: pthread_setattr_default_np: symbol not found

**Comment by roborourke** [comment]
@wupeaking I got that too but it's a separate issue to this one. You should
create a new issue and be more descriptive of the problem e.g. What were
you doing, what did your run & build commands look like etc

**Comment by wupeaking** [comment]
@roborourke https://hub.docker.com/r/ulboralabs/alpine-mongodb/~/dockerfile/  you can try this...

**Comment by ShanikaEdiriweera** [comment]
Hi

I would like to know how to start this container with auth? Or how to setup auth after starting the container?

**Comment by lookat23** [comment]
@ShanikaEdiriweera you can try this
docker run -d --name mongo -p 27017:27017 mvertes/alpine-mongo /root/run.sh mongod --auth --bind_ip 0.0.0.0
